File: examples/weather-mcp/src/main/java/com/example/weather/model/TemperatureUnit.java
/*
 * Copyright (c) 2026 Konstantin Pavlov.
 */

package com.example.weather.model;

import com.fasterxml.jackson.annotation.JsonClassDescription;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonTypeName;

@JsonClassDescription("Unit used to represent temperature.")
@JsonTypeName("TemperatureUnit")
public enum TemperatureUnit {
    @JsonProperty("celsius") CELSIUS,
    @JsonProperty("fahrenheit") FAHRENHEIT
}

File: examples/weather-mcp-kotlin/src/main/kotlin/com/example/weather/model/TemperatureUnit.kt
/*
 * Copyright (c) 2026 Konstantin Pavlov/IT Staff and contributors.
 */

package com.example.weather.model

import kotlinx.serialization.SerialName
import me.kpavlov.kt.schema.Description

@Description("Unit used to represent temperature")
@SerialName("TemperatureUnit")
enum class TemperatureUnit {
    Celsius,
    Fahrenheit,
}
